pyspecdata/lmfitdata.py

```diff
@@ -223,9 +225,11 @@ def eval(self, taxis=None):
             newdata.set_error(self.fit_axis,
                     self.get_error(self.fit_axis))
         # }}}
-        for j,this_name in enumerate(self.pars.keys()):
-            self.pars[this_name].value = p[j]
-        newdata.data[:] = self.run_lambda(self.pars,**{self.fit_axis:taxis}).flatten()
+        for j,this_name in enumerate(self.guess_coeff.keys()):
+            # as discussed, guess_coeff are *not* the fit_coeff.  
+            # (I changed the name from "pars" to "guess_coeff" to clarify that, since "pars" was arguably a bad name!).
+            self.guess_coeff[this_name].value = p[j]
+        newdata.data[:] = self.run_lambda(self.guess_coeff,**{self.fit_axis:taxis}).flatten()
```

Comment:
this is the core issue that needs to be addressed that I'm talking about in the rest of the review.  You are effectively setting the attribute that stores the guess to the fit result, and then running the function using the guess.  Why not just run using the fit result (attribute `fit_coeff`)?